components/core/src/service.rs

```diff
@@ -13,7 +13,7 @@ use std::{fmt,
 
 lazy_static::lazy_static! {
     static ref SG_FROM_STR_RE: Regex =
-        Regex::new(r"\A((?P<application_environment>[^#@]+)#)?(?P<service>[^#@.]+)\.(?P<group>[^#@.]+)(@(?P<organization>[^#@.]+))?\z").unwrap();
+        Regex::new(r"\A((?P<application_environment>[^#@]+)#)?(?P<service>[^#@]+)\.(?P<group>[^#@.]+)(@(?P<organization>[^#@.]+))?\z").unwrap();
```

Comment:
I know you didn't write this regex, but I feel like the presence of named groups makes this significantly more complicated to read than it needs to be.  That's not something you need to change in this PR.  I just wanted my thoughts on this subject noted for the record.

Reply:
really? you find this difficult to read?!! :)

Reply:
TO͇̹̺ͅƝ̴ȳ̳ TH̘Ë͖́̉ ͠P̯͍̭O̚​N̐Y̡ H̸̡̪̯ͨ͊̽̅̾̎Ȩ̬̩̾͛ͪ̈́̀́͘ ̶̧̨̱̹̭̯ͧ̾ͬC̷̙̲̝͖ͭ̏ͥͮ͟Oͮ͏̮̪̝͍M̲̖͊̒ͪͩͬ̚̚͜Ȇ̴̟̟͙̞ͩ͌͝"

Reply:
In all seriousness, since we're making a change; this would be an appropriate time to revise this.

Reply:
I replaced the regex with straight up rust parsing and I don't think that made things better (either in readability or fragility) so I' am not including it here. While like many, my gut reaction is to vomit in the face of regex, this regex really is not that bad. @raskchanky made a comment offline that the readability issue is largely affected by the named groups. I removed the names in favor of unnamed indexed groups which certainly decreases the length of the regex dramatically. One may argue that using numeric indexes adds another level of obscurity. We could use constants to mitigate that but I personally favor named groups over the creation of constants.

Reply:
I dont know if it would be the right tool for the job in this case, but [nom](https://github.com/Geal/nom) is a pretty cool parse combinator I have used in the past. It might be worth looking into.

Reply:
@davidMcneil My gut says that is probably more than we need in this case, but that is indeed cool tech 😄 

I just came across https://bodil.lol/parser-combinators/ recently, so this has coincidentally been on my mind 😆